src/sqlfluff/rules/aliasing/AL09.py

```diff
@@ -60,11 +60,15 @@ def _eval(self, context: RuleContext) -> EvalResultType:
 
         children: Segments = FunctionalContext(context).segment.children()
 
-        for clause_element in children.select(sp.is_type("select_clause_element")):
-            clause_element_raw_segments = (
-                clause_element.get_raw_segments()
-            )  # col_a as col_a
+        # The dialect supports case insensitive naming for a column name, (e.g. COL1
+        # is the same as col1) but the value in the result set name is case sensitive.
+        # We use this to replace the column reference with the alias instead of just
+        # removing the alias.
+        dialect_supports_ci_names_cs_result = ["tsql", "bigquery"]
```

Comment:
I think this comment is hard to follow. Perhaps this would be clearer:
```suggestion
        # Some dialects are case insensitive when referencing columns (e.g. COL1 is
        # the same as col1), but the case of the reference does follow through into
        # the result set (i.e. where COL1 and col1 refer to the same column, the column
        # label in the result set will be different). In these cases, if the alias and
        # the column compare equal (case insensitive), but have a different case, we
        # replace the reference with the alias to achieve a simpler but consistent
        # result.
        dialect_supports_ci_names_cs_result = ["tsql", "bigquery"]
```